### PR TITLE
Refactor Resource View

### LINKED ISSUE #319: Refactor Resource View
implements #300

### PR DESCRIPTION
**Pull request by kenjinp**
fixes #319

### CODE REVIEW

File: src/shared/views/ResourceView.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  const { match, goToOrg, goToProject } = props;
+  const nexus = useNexusContext();
+
+  const {
+    params: { org: orgLabel, project: projectLabel, resourceId },
+  } = match;
+
+  const [{ busy, resource, error }, setResource] = React.useState<{
+    busy: boolean;
+    // TODO remove once SDK is updated to allow generic Resource Types
```
Comment (julienmachon) [inline src/shared/views/ResourceView.tsx]:
```suggestion
    // TODO: remove once SDK is updated to allow generic Resource Types
```

**Comment by kenjinp** [comment]
Please note this won't build until the new version of the SDK is released and integrated

ResourceView has been reimplemented with a couple of changes: 

- metadataCard no longer displays image previews (waiting on https://github.com/BlueBrain/nexus-web/issues/321 )
- metadataCard and some other components have different looks to keep things simple
- lots of small improvements to navigation by url

File: src/shared/components/MetadataCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import { Card, Button, Tooltip, Divider, Descriptions } from 'antd';
+import { Resource } from '@bbp/nexus-sdk';
+
+import TypesIcon from '../Types/TypesIcon';
+import Copy from '../Copy';
+import { getUsername, getResourceLabel } from '../../utils';
+
+import './MetadataCard.less';
+
+const MetadataCardComponent: React.FunctionComponent<{
```
Comment (julienmachon) [inline src/shared/components/MetadataCard/index.tsx]: My issue with this component is its name... `MetadataCard` of what? You basically don't know until you look at the code. Can we rename to something like `ResourceCard` or `ResourceMetadataCard` ?

File: src/shared/components/Types/TypesIcon.tsx [hunk 3]
```diff
@@ -1,8 +1,7 @@
 import * as React from 'react';
-import { Avatar, Tooltip, Tag } from 'antd';
-import * as Identicon from 'identicon.js';
+import { Tooltip, Tag } from 'antd';
 import './Types.less';
-import * as md5 from 'md5';
```
Comment (julienmachon) [inline src/shared/components/Types/TypesIcon.tsx]: is this gone from the dependencies if now unused?

File: src/shared/components/Types/TypesIcon.tsx [hunk 4]
```diff
@@ -1,8 +1,7 @@
 import * as React from 'react';
-import { Avatar, Tooltip, Tag } from 'antd';
-import * as Identicon from 'identicon.js';
```
Comment (julienmachon) [inline src/shared/components/Types/TypesIcon.tsx]: is this gone from the dependencies if now unused?

File: src/shared/containers/ResourceActions.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import { Resource } from '@bbp/nexus-sdk';
+import { useNexusContext } from '@bbp/react-nexus';
+
+import ResourceActions from '../components/ResourceActions';
+import { getResourceLabelsAndIdsFromSelf, getResourceLabel } from '../utils';
+import { notification } from 'antd';
```
Comment (julienmachon) [inline src/shared/containers/ResourceActions.tsx]: format comment, can you move this to the top of the file :pray:

File: src/shared/store/reducers/nexus.ts [hunk 6]
```diff
@@ -62,12 +64,6 @@ export default function nexusReducer(
       activeResource: resourceReducer(state.activeResource, action),
```
Comment (julienmachon) [inline src/shared/store/reducers/nexus.ts]: Can this go away too?

File: src/shared/utils/nexus-maybe.ts [hunk 7]
```diff
@@ -1,33 +1,38 @@
-import { Resource } from '@bbp/nexus-sdk-legacy';
```
Comment (julienmachon) [inline src/shared/utils/nexus-maybe.ts]: Can we rename this file `nexusMaybe` for consistency sake :sake:

File: src/shared/components/MetadataCard/index.tsx (hunk 2 above)
Comment (kenjinp) [reply src/shared/components/MetadataCard/index.tsx]: I like ResourceCard :)

File: src/shared/store/reducers/nexus.ts (hunk 6 above)
Comment (kenjinp) [reply src/shared/store/reducers/nexus.ts]: it turns out, yes :)

**Pull request merged**